pkg/provisioner/ironic/raid/raid.go

```diff
@@ -0,0 +1,162 @@
+func SetTargetRAIDCfg(client *gophercloud.ServiceClient, ironicNode *nodes.Node, raid *metal3v1alpha1.RAIDConfig) (err error) {
+	// Build target for RAID configuration steps
+	logicalDisks, err := buildTargetRAIDCfg(raid)
+	if len(logicalDisks) == 0 || err != nil {
+		return
+	}
+	// Set target for RAID configuration steps
+	return nodes.SetRAIDConfig(
+		client,
+		ironicNode.UUID,
+		nodes.RAIDConfigOpts{LogicalDisks: logicalDisks},
+	).ExtractErr()
+}
+
+// A private method to build RAID disks
+func buildTargetRAIDCfg(raid *metal3v1alpha1.RAIDConfig) (logicalDisks []nodes.LogicalDisk, err error) {
+	if raid != nil {
+		// build logicalDisks
+		if len(raid.HardwareRAIDVolumes) != 0 {
+			logicalDisks, err = buildTargetHardwareRAIDCfg(raid.HardwareRAIDVolumes)
+		} else if len(raid.SoftwareRAIDVolumes) != 0 {
+			logicalDisks, err = buildTargetSoftwareRAIDCfg(raid.SoftwareRAIDVolumes)
+		}
+		// set root volume
+		if len(logicalDisks) != 0 {
+			logicalDisks[0].IsRootVolume = new(bool)
+			*logicalDisks[0].IsRootVolume = true
+		}
+	}
+
+	return
+}
+
+// A private method to build hardware RAID disks
+func buildTargetHardwareRAIDCfg(volumes []metal3v1alpha1.HardwareRAIDVolume) (logicalDisks []nodes.LogicalDisk, err error) {
+	var (
+		logicalDisk    nodes.LogicalDisk
+		nameCheckFlags map[string]struct{} = make(map[string]struct{})
+	)
+
+	for _, volume := range volumes {
+		// Check volume's name
+		if volume.Name != "" {
+			_, exist := nameCheckFlags[volume.Name]
+			if exist {
+				err = errors.Errorf("name of volume must be unique")
+				return nil, err
+			}
+			nameCheckFlags[volume.Name] = struct{}{}
+		}
+		// Build logicalDisk
+		logicalDisk = nodes.LogicalDisk{
+			SizeGB:     volume.SizeGibibytes,
+			RAIDLevel:  volume.Level,
+			VolumeName: volume.Name,
+		}
+		if volume.Rotational != nil {
+			if *volume.Rotational {
+				logicalDisk.DiskType = nodes.HDD
+			} else {
+				logicalDisk.DiskType = nodes.SSD
+			}
+		}
+		if volume.NumberOfPhysicalDisks != nil {
+			logicalDisk.NumberOfPhysicalDisks = *volume.NumberOfPhysicalDisks
+		}
+		// Add to logicalDisks
+		logicalDisks = append(logicalDisks, logicalDisk)
+	}
+
+	return
+}
+
+// A private method to build software RAID disks
+func buildTargetSoftwareRAIDCfg(volumes []metal3v1alpha1.SoftwareRAIDVolume) (logicalDisks []nodes.LogicalDisk, err error) {
+	var (
+		logicalDisk    nodes.LogicalDisk
+		nameCheckFlags map[string]struct{} = make(map[string]struct{})
+	)
+
+	for _, volume := range volumes {
+		// Check volume's name
+		if volume.Name != "" {
+			_, exist := nameCheckFlags[volume.Name]
+			if exist {
+				err = errors.Errorf("name of volume must be unique")
+				return nil, err
+			}
+			nameCheckFlags[volume.Name] = struct{}{}
+		}
+		// Build logicalDisk
+		logicalDisk = nodes.LogicalDisk{
+			SizeGB:     volume.SizeGibibytes,
+			RAIDLevel:  volume.Level,
+			VolumeName: volume.Name,
+			Controller: "software",
+		}
+		/* TODO: gophercloud v0.12.0
```

Comment:
@zaneb The PR about Modify `PhysicalDisks`'s type has been merged, but i can't use it  immediately. Do you have any idea? Thanks.

Reply:
When will be their next release? I can't seem to be able to find any documentation on their release model.

Reply:
According to the date of [here](https://github.com/gophercloud/gophercloud/blob/master/CHANGELOG.md), it should be next month.

Reply:
I believe we have everything ready. This just needs a final look from @zaneb and we should be wrapping up here.